app/components/channel_drawer/channels_list/list/list.js

```diff
@@ -302,83 +209,113 @@ class List extends Component {
         );
     };
 
-    renderDivider = (styles, marginLeft) => {
+    renderSectionSeparator = () => {
+        const {styles} = this.props;
         return (
-            <View
-                style={[styles.divider, {marginLeft}]}
-            />
+            <View style={[styles.divider]}/>
         );
     };
 
     renderItem = ({item}) => {
-        if (!item.isTitle) {
-            return this.createChannelElement(item);
-        }
-        return item.title;
+        return (
+            <ChannelItem
+                channelId={item}
+                onSelectChannel={this.onSelectChannel}
+            />
+        );
     };
 
-    renderTitle = (styles, id, defaultMessage, action, topDivider, bottomDivider) => {
-        const {formatMessage} = this.props.intl;
+    renderUnreadItem = ({item}) => {
+        return (
+            <ChannelUnreadItem
+                channelId={item}
+                onSelectChannel={this.onSelectChannel}
+            />
+        );
+    };
 
-        return {
+    renderSectionHeader = ({section}) => {
+        const {intl, styles} = this.props;
+        const {
+            action,
+            bottomSeparator,
+            defaultMessage,
             id,
-            isTitle: true,
-            title: (
-                <View>
-                    {topDivider && this.renderDivider(styles, 0)}
-                    <View style={styles.titleContainer}>
-                        <Text style={styles.title}>
-                            {formatMessage({id, defaultMessage}).toUpperCase()}
-                        </Text>
-                        {action && this.renderSectionAction(styles, action)}
-                    </View>
-                    {bottomDivider && this.renderDivider(styles, 0)}
+            topSeparator
+        } = section;
+
+        return (
+            <View>
+                {topSeparator && this.renderSectionSeparator()}
+                <View style={styles.titleContainer}>
+                    <Text style={styles.title}>
+                        {intl.formatMessage({id, defaultMessage}).toUpperCase()}
+                    </Text>
+                    {action && this.renderSectionAction(styles, action)}
                 </View>
-            )
-        };
+                {bottomSeparator && this.renderSectionSeparator()}
+            </View>
+        );
     };
 
-    render() {
-        const {styles} = this.props;
+    scrollToTop = () => {
+        if (this.refs.list) {
+            this.refs.list.scrollToOffset({
+                x: 0,
+                y: 0,
+                animated: true
+            });
+        }
+    };
 
-        const {dataSource, showAbove} = this.state;
+    emitUnreadIndicatorChange = debounce((show) => {
```

Comment:
Shouldn't this be part of state? I don't like going outside of React for this

Reply:
way more efficient this way